contributing/contribute-code.md

```diff
@@ -0,0 +1,209 @@
+This means agreeing on the scope and implementation approach of a change.
+
+**Requirements for a Jira ticket to get consensus:**
+
+  - Formal requirements
+     - The *Title* describes the problem concisely.
+     - The *Description* gives all the details needed to understand the problem or feature request.
+     - The *Component* field is set: Many committers and contributors only focus on certain subsystems of Flink. Setting the appropriate component is important for getting their attention.
+  - There is **agreement** that the ticket solves a valid problem, and that it is a **good fit** for Flink.
+    The Flink community considers the following aspects:
+     - Does the contribution alter the behavior of features or components in a way that it may break previous users’ programs and setups? If yes, there needs to be a discussion and agreement that this change is desirable.
+     - Does the contribution conceptually fit well into Flink? Is it too much of a special case such that it makes things more complicated for the common case, or bloats the abstractions / APIs?
+     - Does the feature fit well into Flink’s architecture? Will it scale and keep Flink flexible for the future, or will the feature restrict Flink in the future?
+     - Is the feature a significant new addition (rather than an improvement to an existing part)? If yes, will the Flink community commit to maintaining this feature?
+     - Does this feature align well with Flink's roadmap and currently ongoing efforts?
+     - Does the feature produce added value for Flink users or developers? Or does it introduce the risk of regression without adding relevant user or developer benefit?
+     - Could the contribution live in another repository, e.g., Apache Bahir or another external repository?
+     - Is this a contribution just for the sake of getting a commit in an open source project (fixing typos, style changes merely for taste reasons)
+  - There is **consensus** on how to solve the problem. This includes considerations such as
+    - API and data backwards compatibility and migration strategies
+    - Testing strategies
+    - Impact on Flink's build time
+    - Dependencies and their licenses
+
+Large or controversial changes might require a [Flink Improvement Proposal (FLIP)](https://cwiki.apache.org/confluence/display/FLINK/Flink+Improvement+Proposals) or a discussion on the [dev mailing list]( {{ site.base }}/community.html#mailing-lists ) to reach agreement or consensus.
+
+Once all requirements for the ticket are met, a committer will assign somebody to the *`Assignee`* field of the ticket to work on it.
+Only committers have the permission to assign somebody.
+
+**Pull requests belonging to unassigned Jira tickets will not be reviewed or merged by the community**.
+
+
+<a name="implement"></a>
+
+### 2. Implement your change
+
+- [Set up a Flink development environment](https://cwiki.apache.org/confluence/display/FLINK/Setting+up+a+Flink+development+environment)
```

Comment:
Add an introduction sentence like "Once you've been assigned to a Jira issue, you may start to implement the required changes."